### PR TITLE
Improve test runner voice commands

### PR DESCRIPTION
**Pull request by pokey**
## Checklist

- [-] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [x] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [-] I have not broken the cheatsheet

### CODE REVIEW

File: cursorless-talon-dev/src/cursorless_dev_vscode.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+mod.list(
+    "cursorless_launch_configuration", "A launch configuration for Cursorless vscode"
+)
+
+ctx.lists["user.cursorless_launch_configuration"] = {
+    "stench": "Run extension",
+    "test": "Extension tests",
+    "test unit": "Unit tests only",
+    "test subset": "Run test subset",
```
Comment (pokey) [inline cursorless-talon-dev/src/cursorless_dev_vscode.py:20]:
```suggestion
    "test subset": "Extension tests subset",
```

File: cursorless-talon-dev/src/cursorless_dev_vscode.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "test": "Extension tests",
+    "test subset": "Extension tests subset",
+    "test unit": "Unit tests only",
+    "test talon": "Talon tests",
+    "test talon subset": "Talon tests subset",
+    # "Update fixtures": "Update fixtures",
+    # "Update fixtures subset": "Update fixtures subset",
+    # "Docusaurus start": "Docusaurus start",
+    # "Docusaurus build": "Docusaurus build",
+    # "cursorless.org client-side": "cursorless.org client-side",
```
Comment (pokey) [inline cursorless-talon-dev/src/cursorless_dev_vscode.py:27]: Some of these seemed to dangerous / niche to map

File: cursorless-talon-dev/src/cursorless_dev_vscode.talon [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 ^debug generate subset$:
-    user.vscode_with_plugin("workbench.action.tasks.runTask", "Generate test subset file")
+    user.run_rpc_command("workbench.action.tasks.runTask", "Generate test subset file")
 ^debug edit subset$:
-    user.vscode_with_plugin("commands.openFolder", "./packages/test-harness/testSubsetGrep.properties")
+    user.run_rpc_command("commands.openFolder", "./packages/test-harness/testSubsetGrep.properties")
+
+debug {user.cursorless_launch_configuration}:
+    user.run_rpc_command("commands.startDebugging", cursorless_launch_configuration)
+    user.run_rpc_command("workbench.debug.action.focusRepl")
```
Comment (pokey) [inline cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: I find I want to see the debug output 95% of the time so this saves me saying "debug console" after "debug test"

File: cursorless-talon-dev/src/cursorless_dev_vscode.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+app: vscode
+win.title: /cursorless/
+"""
+
+
+mod.list(
+    "cursorless_launch_configuration", "A launch configuration for Cursorless vscode"
+)
+
+ctx.lists["user.cursorless_launch_configuration"] = {
```
Comment (josharian) [inline cursorless-talon-dev/src/cursorless_dev_vscode.py:16]: use a `.talon-list`? i find i end up with one eventually, so may as will start there.

File: cursorless-talon-dev/src/cursorless_dev_vscode.talon (hunk 3 above)
Comment (josharian) [reply cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: i find that 95% of the time after starting the extension, i want to switch to that window. unfortunately, doing that is pretty non-portable (afaik).

Comment (josharian) [reply cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: What do you think about hiding the repl on stench stop?

**Review by josharian — APPROVED** [review]
lgtm, but i have my own commands mapped for these, using "stench" as the anchor noun (stench start, stench stop, stench reload, stench unit, etc.) so these won't affect me much

File: cursorless-talon-dev/src/cursorless_dev_vscode.talon (hunk 3 above)
Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: > i find that 95% of the time after starting the extension, i want to switch to that window. unfortunately, doing that is pretty non-portable (afaik).

For me that happens automatically I believe. Is that not the case for you?

Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: > What do you think about hiding the repl on stench stop?

Huh I don't have a strong feeling. Does the debug output get clobbered when you stop? Or is that just when you start another run? If it gets clobbered then we should def hide; if not I could go either way

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 4 above)
Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:16]: Ah interesting idea. Haven't actually played with those yet. I'll try that

Comment (AndreasArvidsson) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:16]: I agree but that would make Cursorless 0.4 only

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 2 above)
Comment (AndreasArvidsson) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:27]: Not sure if I'm a fan of these tors terse launch commands

File: cursorless-talon-dev/src/cursorless_dev_vscode.talon (hunk 3 above)
Comment (AndreasArvidsson) [reply cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: When you start a new I believe

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 4 above)
Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:16]: This is just for Cursorless contributors. I'm comfortable requiring them to be on 0.4

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 2 above)
Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:27]: Most of them mirror the launch name pretty closely. Are you referring to "stench"?

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 4 above)
Comment (AndreasArvidsson) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:16]: Same here

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 2 above)
Comment (AndreasArvidsson) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:27]: To be honest the stench one is a bit unintuitive

Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:27]: would be nice if it were configurable, but I'm not sure how to do that easily. I guess we could just have the slightly more verbose version and contributor can map separate short command if they want it?

Comment (AndreasArvidsson) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:27]: That sounds reasonable.

File: cursorless-talon-dev/src/cursorless_dev_vscode.py (hunk 4 above)
Comment (auscompgeek) [reply cursorless-talon-dev/src/cursorless_dev_vscode.py:16]: For future travellers, we already made Cursorless require Talon 0.4, see #1880.

File: cursorless-talon-dev/src/cursorless_dev_vscode.talon (hunk 3 above)
Comment (pokey) [reply cursorless-talon-dev/src/cursorless_dev_vscode.talon:25]: I don't know how to hide the repl on stopping extension. I'd be inclined to punt on that anyway and get this PR in

**Comment by pokey** [comment]
Ok this one is ready for another review @AndreasArvidsson @josharian

File: docs/contributing/CONTRIBUTING.md [hunk 5]
```diff
@@ -52,17 +52,19 @@ extension](#running--testing-extension-locally). You may also find the [VSCode A
 ## Running / testing extension locally
 
 In order to test out your local version of the extension or to run unit tests
-locally, you need to run the extension in debug mode. To do so you need to run
-the `workbench.action.debug.selectandstart` command in VSCode and then select either "Run
-Extension" or "Extension Tests".
+locally, you need to run the extension in debug mode.
+
+To run the extension, say `"debug extension"` ("stench" is short for "extension" 😊). The run the tests, say `"debug test"`.
```
Comment (AndreasArvidsson) [inline docs/contributing/CONTRIBUTING.md:57]: The comment about `stench` is not needed here anymore

File: .vscode/launch.json
Comment (pokey) [inline .vscode/launch.json:1]: Improved names and changed ordering

File: .vscode/tasks.json
Comment (pokey) [inline .vscode/tasks.json:1]: improved names and changed ordering, as well as improted install local and uninstall local to use our scripts

**Comment by pokey** [comment]
I lifted a cleanup of `launch.json` and `tasks.json` out of #2256 and added them to this PR because they change the names of our launch configs. Feel free to merge if this looks good to you @AndreasArvidsson

**Comment by pokey** [comment]
I also incorporated some of the changes from #2424 to avoid a merge conflict

File: cursorless-talon-dev/src/cursorless_launch_configuration.talon-list [hunk 6]
```diff
@@ -0,0 +1,19 @@
+app: vscode
+win.title: /cursorless/
+list: user.cursorless_launch_configuration
```
Comment (AndreasArvidsson) [inline cursorless-talon-dev/src/cursorless_launch_configuration.talon-list:3]: I'm pretty sure `list:` needs to be on the first line. Have you tested this?

Comment (pokey) [reply cursorless-talon-dev/src/cursorless_launch_configuration.talon-list]: done

**Review by AndreasArvidsson — APPROVED** [review]

**Pull request merged**